addon/mixins/route.js

```diff
@@ -217,6 +216,19 @@ export default Ember.Mixin.create({
     return false;
   },
 
+  /**
+   Update the infinity model with new objects
+
+   @method updateInfinityModel
+   @param {Ember.Enumerable} newObjects The new objects to add to the model
+   @return {Ember.Array} returns the updated infinity model
+   */
+  updateInfinityModel(newObjects) {
+    var infinityModel = this.get(this.get('_modelPath'));
+
+    return infinityModel.pushObjects(newObjects);
+  },
+
```

Comment:
It would work for the majority of developers, unless they've explicitly turned off prototype extensions, and if they have, they're probably going to be aware of that stuff.